sagemaker_huggingface_inference_toolkit/decoder_encoder.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2021 The HuggingFace Team, Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import datetime
+import json
+
+import numpy as np
+from sagemaker_inference import content_types, errors
+
+
+def decode_json(content):
+    return json.loads(content)
+
+
+# https://github.com/automl/SMAC3/issues/453
+class _JSONEncoder(json.JSONEncoder):
+    """
+    custom `JSONEncoder` to make sure float and int64 ar converted
+    """
+
+    def default(self, obj):
+        if isinstance(obj, np.integer):
+            return int(obj)
+        elif isinstance(obj, np.floating):
+            return float(obj)
+        elif isinstance(obj, np.ndarray):
+            return obj.tolist()
+        elif isinstance(obj, datetime.datetime):
+            return obj.__str__()
+        else:
+            return super(_JSONEncoder, self).default(obj)
+
+
+def encode_json(content):
+    """
+    encodes json with custom `JSONEncoder`
+    """
+    return json.dumps(
+        content,
+        ensure_ascii=False,
+        allow_nan=False,
+        indent=None,
+        cls=_JSONEncoder,
+        separators=(",", ":"),
+    )
+
+
+_encoder_map = {
+    # content_types.NPY: _array_to_npy,
+    # content_types.CSV: _array_to_csv,
+    content_types.JSON: encode_json,
+}
+_decoder_map = {
+    # content_types.NPY: _array_to_npy,
```

Comment:
could we keep them as-is or just call Sagemaker-inference's encoder/decoder? Customers should be able to define their custom pre/post processing code.